start

```diff
@@ -45,18 +61,72 @@ cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
 host    all             all             0.0.0.0/0               md5
 EOF
 
+# allow replication of the database
+if [ -n "${REP_USER}" ]; then
+  if [ "${PSQL_SSLMODE}" == "disable" ]; then
+    cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
+host    replication     $REP_USER       0.0.0.0/0               md5
+EOF
+    else 
+    cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
+hostssl replication     $REP_USER       0.0.0.0/0               md5
+EOF
+  fi
+fi
+
+
+# allow passwordless connections from current user
+if [ -n "${DB_USER}" ]; then
+  if [ ! -z "${DB_PASS}" ]; then
+    cat >> ~/.pgpass <<EOF
+*:*:*:${DB_USER}:${DB_PASS}
+EOF
+    chmod 0600 ~/.pgpass
+  fi
+fi
+
+if [ "${PSQL_SLAVE}" == "false" ]; then
+  if [ -n "${REP_USER}" ]; then
+    echo "Supporting hot standby..."
+    cat >> ${PG_CONFDIR}/postgresql.conf <<EOF
+wal_level = hot_standby
+max_wal_senders = 3
+checkpoint_segments = 8
+wal_keep_segments = 8
+EOF
+## NOTES: archive_mode = 'on' should be a supported flag.
+  fi
+fi
+
+
 cd ${PG_HOME}
 
 # initialize PostgreSQL data directory
 if [ ! -d ${PG_DATADIR} ]; then
-  # check if we need to perform data migration
-  PG_OLD_VERSION=$(find ${PG_HOME}/[0-9].[0-9]/main -maxdepth 1 -name PG_VERSION 2>/dev/null | sort -r | head -n1 | cut -d'/' -f5)
+  if [ ! "${PSQL_SLAVE}" == "false" ]; then
+   echo "Replicating database..."
+   echo "${REP_PASS}" | sudo -u postgres -H "${PG_BINDIR}/pg_basebackup" -D "${PG_DATADIR}" \
+   -h "${REP_HOST}" -p "${REP_PORT}" -U "${REP_USER}" -W -v -P
+   echo "Setting up hot standby configuration..."
+   cat >> ${PG_CONFDIR}/postgresql.conf <<EOF
+hot_standby = on
+EOF
+   cat >> ${PG_DATADIR}/recovery.conf <<EOF
```

Comment:
you might want to change the ownership of the `recovery.conf` file, its currently owned by root